Test/Platform/Tools/MFTestSystem/TestSystem.cs

```diff
@@ -79,19 +79,20 @@ static void Main(string[] args)
             m_args = args;
             try
             {
-                m_testSystem.RunTests();
+                return m_testSystem.RunTests() ? 0 : -1;
             }
             catch (Exception ex)
             {
                 Utils.WriteToEventLog(String.Format("TestSystem: " + 
                     "An uncaught exception was thrown when invoking RunTests: {0}",ex.ToString()));
+                return -1;
             }
         }
 
         /// <summary>
         /// Use this method to execute tests.
         /// </summary>
-        public void RunTests()
+        public bool RunTests()

```

Comment:
Avoid using bool for success or failure.  Use error codes or exceptions.  In this case, we should throw. Main can catch and exit the process with an error code.